src/containers/Authentication/Signup.js

```diff
@@ -0,0 +1,37 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { signup } from '../../redux/actions/authActions';
+
+class Signup extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    let email, password, f_name, l_name;
+    const { dispatch } = this.props;
+    return (
+      <form
+        onSubmit={e => {
+          dispatch(
+            signup({
+              email: email.value,
+              password: password.value,
+              f_name: f_name.value,
+              l_name: l_name.value
+            })
+          );
+        }}
+      >
+        Signup:
+        <input ref={node => (email = node)} />
+        <input ref={node => (password = node)} />
+        <input ref={node => (f_name = node)} />
+        <input ref={node => (l_name = node)} />
```

Comment:
I may be wrong, but I believe it's poor form to be using refs for each input. The idea behind react is that there is one-way data flow from a parent element down to a child element, and from my understanding, ref kind of inverts that. What's happening here is you're making the text from the children inputs available to the "parent" form, rather than storing the values for the inputs in the form's state and binding the two. See this link for more information: [https://stackoverflow.com/questions/23427384/get-form-data-in-reactjs](url).

One thing I like about your approach is that it's very concise. However, it's not as standard and perhaps less scalable than the traditional react data flow.

Reply:
I haven't looked into Reacts best practices yet. I just built this form out to test the stores functionality, but this makes sense. I'll look into improving upon this for us to potentially be able to use this page as our official signup page. Will be reading up on React later today after I've made the necessary changes to the store.